### PR TITLE
[STLND-687]

### PR DESCRIPTION
**Pull request by jaszczur171**
- Added support for checking queue state for isEmpty and retreiving the size of the queue.
- Provided a test confirming correct working of the new methods.

[STLND-656]
- Updated StellaNowMessageWrapper Tests to conform with the new message wrapper expectations.

### CODE REVIEW

**Review by badsector3 — APPROVED** [review]
LGTM

File: StellaNowSDKDemo/src/Messages/UserLoginMessage.cs [hunk 1]
```diff
@@ -12,42 +12,48 @@ namespace StellaNowSDKDemo.Messages;
 
 public class UserLoginMessage : StellaNowMessageWrapper
 {
-    public UserLoginMessage(string userId, DateTime timestamp, string patronId)
+    public UserLoginMessage(string punterId, DateTime timestamp, int userGroupId, string userId)
```
Comment (sebtom) [inline StellaNowSDKDemo/src/Messages/UserLoginMessage.cs:15]: shall we keep the test project separated from SDK? changes in generated classes should not trigger release (now that we have CICD and builds from master branch). WDYT?

File: StellaNowSDKDemo/src/Program.cs [hunk 2]
```diff
@@ -61,6 +61,12 @@ static async Task Main(string[] args)
             stellaSdk.SendMessage(message);
             logger.LogInformation("New Message Queued");
 
+            logger.LogInformation(
```
Comment (sebtom) [inline StellaNowSDKDemo/src/Program.cs:64]: I do not understand how this works? this loop iterates up to `MessageCount` which is private and it's always 1. do you change this variable if you want to run test with eg 100 events? maybe it should be passed as `string[] args` argument?

File: StellaNowSDKDemo/src/Program.cs [hunk 3]
```diff
@@ -61,6 +61,12 @@ static async Task Main(string[] args)
...
+                "Queue has messages {HasMessages} and count is {Count}",
+                stellaSdk.HasMessagesPendingForDispatch(),
+                stellaSdk.MessagesPendingForDispatchCount()
+            );
+
```
Comment (sebtom) [inline StellaNowSDKDemo/src/Program.cs:69]: maybe as we have example program now, we should make use of `HasMessagesPendingForDispatch()` method and add a loop around `// Disconnect and terminate` and wait for all messages to be transmitted before we kill?

File: StellaNowSDKTests/src/Services/.gitkeep
Comment (sebtom) [inline StellaNowSDKTests/src/Services/.gitkeep:1]: why do you need gitkeep is this folder is not empty?

File: StellaNowSDKTests/src/Services/StellaNowSdkTests.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                _mockConnectionStrategy.Object, 
+                _mockMessageQueue.Object, 
+                new StellaNowConfig() // use default configuration
+            );
+        }
+
+        [TestMethod]
+        public void Test_HasMessagesPendingForDispatch()
+        {
+            // Setup the mock to return true when IsQueueEmpty is called
```
Comment (sebtom) [inline StellaNowSDKTests/src/Services/StellaNowSdkTests.cs:37]: well, this line setup something opposite than this comments says ;)

File: StellaNowSDKDemo/src/Program.cs (hunk 3 above)
Comment (sebtom) [reply StellaNowSDKDemo/src/Program.cs:69]: ok I see it's done in https://github.com/stella-systems/stellanow-sdk-csharp/pull/7/files, right?

File: StellaNowSDKTests/src/Services/StellaNowSdkTests.cs (hunk 4 above)
Comment (jaszczur171) [reply StellaNowSDKTests/src/Services/StellaNowSdkTests.cs:37]: Copy and pastę fail

File: StellaNowSDKTests/src/Services/.gitkeep
Comment (jaszczur171) [reply StellaNowSDKTests/src/Services/.gitkeep:1]: It's a left over, I'll clean this

**Pull request merged**